### PR TITLE
Feature/streaming insert

### PR DESCRIPTION
**Pull request by dmfenton**
Part of #31

### CODE REVIEW

File: models/Socrata.js [hunk 1]
```diff
@@ -93,122 +90,177 @@ var Socrata = function (koop) {
...
+        callback(null, [{status: 'processing'}])
+        // Simultaneously gather count and metadata by adding them to the queue with their variables
+        infoQueue.push({
+            task: socrata.getRowCount,
+            host: host,
+            urlId: urlId
+        }, function (err, rowCount) {
+              if (err) {
+                // set status to failed
+                return
```
Comment (dmfenton) [inline models/Socrata.js]: What's the appropriate thing to do here? This is hanging the response when this request fails.

Comment (chelm) [inline models/Socrata.js]: @dmfenton I dont know enough about all the changes to understand but I'll pull it down in a bit an look and we hang to talk about it.

**Comment by dmfenton** [comment]
Still need to rebuild the check cache method and add some tests.

**Comment by dmfenton** [comment]
Hey @chelm or @ngoldman review and merge por favor?

**Comment by chelm** [comment]
I'll merge it cause its pretty awesome!

**Pull request merged**